### PR TITLE
New features for main branch and eventual release

### PR DESCRIPTION
**Pull request by bmeyers**
This PR will contain new features that are added to support the following ongoing tasks:

- validation of tilt/azimuth estimation (PVSC paper)
- validation of time-shift detector (PVRW poster)
- Odd and ends related to OSD translation (see PR #90) 
-  bug fixes in @claireberschauer's PVPro post-processor module
- clean up of some imports and requirements

### CODE REVIEW

**Comment by bmeyers** [comment]
I had to adjust the tests for latitude, longitude, tilt, and estimation based on the changes I made while working on the PVSC paper. This is going to cause changes in results, so this will be a minor release, `0.8.0`.

**Comment by bmeyers** [comment]
a couple final comments:

- Apparently `np.testing.assert_array_almost_equal` has been depreciated in favor of `np.testing.assert_allclose`, which has a more clearly defined floating point arithmetic. The threshold is set differently, and i just used the absolute tolerance or `atol` in these test. [See the documentation](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html) for more detail.
- Note that the logic in `tests/solardatatools/test_system_profiler.py` is actually different now. We're now comparing values calculated by the code to what that code should produce at a specific moment in time. Before, the tests weren't actually correct unit tests. My apologies for not catching that sooner.

File: pvsystemprofiler/utilities/hour_angle_equation.py [hunk 1]
```diff
@@ -6,18 +6,19 @@
 from pvsystemprofiler.utilities.time_convert import clock_to_solar
 
 
-def calculate_omega(data_sampling, num_days, lon, doy, gmt_offset):
+def calculate_omega(doy, data_sampling, lon, gmt_offset):
```
Comment (pluflou) [inline pvsystemprofiler/utilities/hour_angle_equation.py:9]: Looks like this is still called once in [tests/fixtures/tilt_azimuth/tilt_azimuth_estimation_data_creator.ipynb](https://github.com/slacgismo/solar-data-tools/blob/0a8c2db4d12f37fd684abe19f247f9e36b60e0a8/tests/fixtures/tilt_azimuth/tilt_azimuth_estimation_data_creator.ipynb) using the previous order of params. Should we update it or was that notebook just used once for creating data for tests?

File: pvsystemprofiler/algorithms/performance_model_estimation.py [hunk 2]
```diff
@@ -6,23 +6,54 @@
 """
 
 
-def find_fit_costheta(data_matrix, clear_index):
+def find_fit_costheta(data_matrix, clear_index, doy):
     """
     :param data_matrix: power matrix.
     :param clear_index: boolean array specifying clear days.
```
Comment (pluflou) [inline pvsystemprofiler/algorithms/performance_model_estimation.py:12]: We should add the doy param here. Is it also as below?
```suggestion
    :param doy: day of year (float or array)
```

File: pvsystemprofiler/tilt_azimuth_study.py [hunk 3]
```diff
@@ -132,10 +162,13 @@ def run(self, delta_method=('cooper', 'spencer')):
...
         delta_method = np.atleast_1d(delta_method)
         # calculate hour angle
-        self.omega = calculate_omega(self.data_sampling, self.num_days, self.lon_input, self.day_of_year,
-                                     self.gmt_offset)
+        self.omega = calculate_omega(
+            self.day_of_year, self.data_sampling, self.lon_input, self.gmt_offset
+        )
         # fit daily signal of cos theta
-        self.scale_factor_costheta, self.costheta_fit = find_fit_costheta(self.data_matrix, self.clear_index)
+        self.scale_factor_costheta, self.costheta_fit = find_fit_costheta(
```
Comment (pluflou) [inline pvsystemprofiler/tilt_azimuth_study.py:169]: @bmeyers Looks like this is missing the `doy` parameter, and the returned values are different now as well (although I'm not certain how that would change this function), based on the changes in [pvsystemprofiler/algorithms/performance_model_estimation.py](https://github.com/slacgismo/solar-data-tools/pull/91/files#diff-f58829f1c72480217b9bd0318c96c61a7f773e80405ad07512d0ae618351db43).

**Review by pluflou — COMMENTED** [review]
@bmeyers I added a couple of comments, the last one [here](https://github.com/slacgismo/solar-data-tools/pull/91/files#r1151233788) being the most important one in terms of functionality.

File: pvsystemprofiler/utilities/hour_angle_equation.py (hunk 1 above)
Comment (bmeyers) [reply pvsystemprofiler/utilities/hour_angle_equation.py:9]: I should take a look at that notebook to make sure. thanks for finding that.

File: pvsystemprofiler/algorithms/performance_model_estimation.py [hunk 4]
```diff
@@ -6,23 +6,55 @@
 """
 
 
-def find_fit_costheta(data_matrix, clear_index):
+def find_fit_costheta(data_matrix, clear_index, doy):
```
[the remark this answers is not in the document]
Comment (bmeyers) [reply pvsystemprofiler/algorithms/performance_model_estimation.py:9]: done in most recent commit

**Pull request merged**